ChiModules/KEigenvalueSolver/IterativeMethods/power_iteration.cc

```diff
@@ -25,161 +28,128 @@ using namespace LinearBoltzmann;
-      double rho = sqrt(pw_change/pw_change_prev);
-      pw_change_prev = pw_change;
-      nit += 1;
-
-      if (si_nit==0) rho = 0.0;
-      if (pw_change<std::max(groupset.residual_tolerance*rho,1.0e-10))
-        si_converged = true;
-
-      // ----- Print iteration information
-      std::string offset = "    ";
-      std::stringstream si_iter_info;
-      si_iter_info
-        << chi_program_timer.GetTimeString() << " "
-        << offset
-        << "WGS groups ["
-        << groupset.groups.front().id
-        << "-"
-        << groupset.groups.back().id
-        << "]"
-        << " Source Iteration " << std::setw(5) << si_nit
-        << " Point-wise change " << std::setw(14) << pw_change;
-        
-      if (si_converged)
-        si_iter_info << " CONVERGED\n";
-      chi_log.Log(LOG_0VERBOSE_1) << si_iter_info.str();
-
-      if (si_converged) {
-        chi_log.Log(LOG_0VERBOSE_1)
-            << "\nSource iterations converged in "
-            << si_nit << " iteratrions.\n";
-        break;
-      }
-    }//for source iterations
-
-    if (!si_converged) {
-      chi_log.Log(LOG_ALLVERBOSE_1)
-          << "\n!!!WARNING!!! "
-             "Source iterations did not converge.\n";
+      ClassicRichardson(groupset, 0, sweep_scheduler,
+                        APPLY_WGS_SCATTER_SOURCE | APPLY_AGS_SCATTER_SOURCE,
+                        options.verbose_inner_iterations);
+    }
+    else if (groupset.iterative_method == IterativeMethod::GMRES)
+    {
+      GMRES(groupset, 0, sweep_scheduler,
+            APPLY_WGS_SCATTER_SOURCE, APPLY_AGS_SCATTER_SOURCE,
+            options.verbose_inner_iterations);
     }
 
-    // ----- Recompute eigenvalue
+    //============================== Recompute k-eigenvalue
     double F_new = ComputeProduction();
-    k_eff = F_new/F_prev * k_eff;
+    k_eff = F_new / F_prev * k_eff;
     double reactivity = (k_eff - 1.0) / k_eff;
 
-    // ----- Compute convergence parameters and bump values
-    double k_eff_change = fabs(k_eff - k_eff_prev) / k_eff;
-    k_eff_prev = k_eff; F_prev = F_new;
+    //============================== Check convergence, reset book-keeping
     ScopedCopySTLvectors(groupset, phi_new_local, phi_prev_local);
-                                          
-    if (k_eff_change<std::max(options.tolerance, 1.0e-12))
-      k_converged = true;    
-
-    // ----- Print iteration summary
-    std::stringstream k_iter_info;
-    k_iter_info
-      << chi_program_timer.GetTimeString() << " "
-      << "  Iteration " << std::setw(5) << nit
-      << "  k_eff " << std::setw(10) << k_eff
-      << "  k_eff change " << std::setw(10) << k_eff_change
-      << "  reactivity " << std::setw(10) << reactivity * 1e5;
-    if (k_converged) {
-      k_iter_info << " CONVERGED\n";
+    double k_eff_change = fabs(k_eff - k_eff_prev) / k_eff;
+    k_eff_prev = k_eff;
+    F_prev = F_new;
+    nit += 1;
+
+    if (k_eff_change < std::max(options.tolerance, 1.0e-12))
```

Comment:
Within the ClassicRichardson method this is used: 

`if (pw_change<std::max(groupset.residual_tolerance*(1.0-rho),1.0e-10))`

which was the primary reason this has been here since the adoption of the `KEigenvalue::Solver` module. Perhaps make an issue for this to be addressed later.